### DISCUSSION TITLE
Save changes in combinations should Save when clicking on next combination button

### DISCUSSION THREAD

**Discussion by LinkOfLight** [PrestaShop/PrestaShop#35272] [Ideas]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

On the new product page, when you click on modify a combination before you could for example assign a color to the combination and this change was saved by simply clicking on the next combination button, now you need to hit save to be able to move on to combination, making this task much slower and heavier

### Expected behavior

Save the changes made to the combination by clicking on the next combination button

### Steps to reproduce

1. go to a product whit combinations
2. Click on Modify Combination
3. Make a change
4. Push in next combination
5. Dont save the change X

### PrestaShop version(s) where the bug happened

8.1.3

### PHP version(s) where the bug happened

8.1

**Comment by RosaBenouamer** [comment]
@TristanLDD

**Comment by kpodemski** [comment]
Maybe there should be a button explicitly for "Save & Continue", or "Save & Go to the next combination"? I'm not sure if saving something because you clicked "Next combination" button is a good idea.

**Comment by ElodieOS** [comment]
In my opinion, we need to display a modal if the user doesn't save before clicking on the next combination button, but we need to change the message. 

Today we have: "All unsaved modifications made on the combination XXX will be lost. Cancel - Discard" 

<img width="1371" alt="Capture d’écran 2024-05-21 à 11 17 22" src="https://github.com/PrestaShop/PrestaShop/assets/113457393/9e52f391-bbac-4149-919d-66b8be3e6053">

I think it will be better to display something like: "Do you want to save your changes before moving on to the next combination? Discard - Save and continue

Wdyt @Julievrz?

**Comment by Julievrz** [comment]
@ElodieOS I think it's a good idea. We could have the following content:

Title: Save and continue? 

Body message: Save your changes on the **%combination%** combination  before moving on to the next one.

CTAs: Discard changes / Save and continue
